include/drivers/pwm.h

```diff
@@ -186,6 +406,176 @@ static inline int pwm_pin_set_nsec(const struct device *dev, uint32_t pwm,
 				  (uint32_t)pulse_cycles, flags);
 }
 
+/**
+ * @brief Convert from PWM cycles to microseconds.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param pwm PWM pin.
+ * @param cycles Cycles to be converted.
+ * @param usec Pointer to the memory to store calculated usec.
+ *
+ * @retval 0 If successful.
+ * @retval -EIO If cycles per second cannot be determined.
+ * @retval -ERANGE If result is too large.
+ */
+static inline int pwm_pin_cycles_to_usec(const struct device *dev, uint32_t pwm,
+					 uint32_t cycles, uint64_t *usec)
+{
+	uint64_t cycles_per_sec;
+	uint64_t temp;
+
+	if (pwm_get_cycles_per_sec(dev, pwm, &cycles_per_sec) != 0) {
+		return -EIO;
+	}
+
+	if (u64_mul_overflow(cycles, (uint64_t)USEC_PER_SEC, &temp)) {
+		return -ERANGE;
+	}
+
+	*usec = temp / cycles_per_sec;
+
+	return 0;
+}
+
+/**
+ * @brief Convert from PWM cycles to nanoseconds.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param pwm PWM pin.
+ * @param cycles Cycles to be converted.
+ * @param nsec Pointer to the memory to store the calculated nsec.
+ *
+ * @retval 0 If successful.
+ * @retval -EIO If cycles per second cannot be determined.
+ * @retval -ERANGE If result is too large.
+ */
+static inline int pwm_pin_cycles_to_nsec(const struct device *dev, uint32_t pwm,
+					 uint32_t cycles, uint64_t *nsec)
+{
+	uint64_t cycles_per_sec;
+	uint64_t temp;
+
+	if (pwm_get_cycles_per_sec(dev, pwm, &cycles_per_sec) != 0) {
+		return -EIO;
+	}
+
+	if (u64_mul_overflow(cycles, (uint64_t)NSEC_PER_SEC, &temp)) {
+		return -ERANGE;
+	}
+
+	*nsec = temp / cycles_per_sec;
+
+	return 0;
+}
+
+/**
+ * @brief Capture a single PWM period/pulse width in microseconds for a single
+ *        PWM input.
+ *
+ * This API function wraps calls to @a pwm_pin_capture_cycles() and @a
+ * pwm_pin_cycles_to_usec() and passes the capture results to the caller. The
+ * function is blocking until either the PWM capture is completed or a timeout
+ * occurs.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param pwm PWM pin.
+ * @param flags PWM capture flags.
+ * @param period Pointer to the memory to store the captured PWM period width
+ *               (in usec).
+ * @param pulse Pointer to the memory to store the captured PWM pulse width (in
+ *              usec).
+ * @param timeout Waiting period for the capture to complete.
+ *
+ * @retval 0 If successful.
+ * @retval -EBUSY PWM capture already in progress.
+ * @retval -EAGAIN Waiting period timed out.
+ * @retval -EIO IO error while capturing.
+ * @retval -ERANGE If result is too large.
+ */
+
```

Comment:
unnecessary blank line?  several occurrences in file.

Reply:
Fixed, thanks.